test_elasticsearch/test_connection.py

```diff
@@ -32,6 +32,10 @@ def test_ssl_context(self):
         )
         self.assertTrue(con.use_ssl)
 
+    def test_http_compression(self):
+        con = Urllib3HttpConnection(http_compress=True)
+        self.assertTrue(con.http_compress)
```

Comment:
we should also verify that `con.headers` are set properly, once the headers are set in `__init__`.